collections/schema.go

```diff
@@ -0,0 +1,52 @@
+package collections
+
+import (
+	"fmt"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"regexp"
+)
+
+// NewSchema creates a new schema from the provided store key.
+func NewSchema(storeKey storetypes.StoreKey) Schema {
+	return Schema{
+		storeKey:            storeKey,
+		collectionsByName:   map[string]Collection{},
+		collectionsByPrefix: map[string]Collection{},
+	}
+}
+
+// Schema specifies a group of collections stored within the storage specified
+// by a single store key. All the collections within the schema must have a
+// unique binary prefix and human-readable name. Schema will eventually include
+// methods for importing/exporting genesis data and for schema reflection for
```

Comment:
there was some mention in the past that some users don't want to expose all data in the db via a query. I feel like a autogenerated client here goes against that, no? we might need to do some user research before doing something like that?

Reply:
I guess it's kind of mute with abci_query.

Reply:
I think that's more of an access permission thing which sits at a level up. All queries are public (unless we get into supporting private blockchains). But a node operator may want to restrict certain paths. We would need to build that infrastructure for all queries or a user can do that with a proxy server